### DISCUSSION TITLE
Custom HID device issues

### DISCUSSION THREAD

**Discussion by wevfreeman** [sekigon-gonnoc/Pico-PIO-USB#71] [Q&A]
Hello! I found your repository while researching ways to create a custom HID device for my undergraduate capstone project. I'm at a loss for how to debug this. I tried following the example `usb_device.c`, but I'm not seeing results. I've defined all of my descriptors according to the HID spec, but the device is not enumerating properly. Would anyone be able to walk me through this?

**Comment by ccrome** [comment]
Are you sure your USB connections are correct?  Describe which pins are connected to which.

For a USB device:  connect usb connector D+ to pin GP0.  D- to pin GP1, and additionally, put a 1.5kOhm pull up resistor between D+ and 3.3V (not VBUS).  If you're powering the pico from the USB device, you also need to connect VBUS on the connector to VSYS on the pico.
